activerecord/lib/active_record/associations/association.rb

```diff
@@ -331,7 +331,10 @@ def skip_statement_cache?(scope)
         end
 
         def enqueue_destroy_association(options)
-          owner.class.destroy_association_async_job&.perform_later(**options)
+          job_class = owner.class
+          ActiveRecord::Base.transactional_job_enqueue options do
```

Comment:
Would not this add an after commit callback to all the models in the application even if they don't have destroy async associations?